client/src/featureform/resources.py

```diff
@@ -277,6 +276,22 @@ def _create_local(self, db) -> None:
                   str(self.config.serialize(), 'utf-8')
                   )
 
+    def __eq__(self, other):
+        if self.name != other.name:
+            return False
+        if self.description != other.description:
+            return False
+        if self.team != other.team:
+            return False
+        if str(self.config.serialize(), 'utf-8') != str(other.config.serialize(), 'utf-8'):
+            return False
+        if self.config.type() != other.config.type():
+            return False
+        if self.config.software() != other.config.software():
+            return False
+        return True
```

Comment:
there are ways of cycling though all the attributes of a class iteratively rather than hardcoding every comparison
https://stackoverflow.com/questions/11637293/iterate-over-object-attributes-in-python
stuff like this would be cleaner, and would also mean these methods wouldn't need to be updated every time an attribute is added to a class